File: app/src/main/java/com/kkkkkn/readbooks/view/activities/BaseActivity.java
package com.kkkkkn.readbooks.view.activities;

import android.graphics.Color;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.ViewCompat;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleEventObserver;
import androidx.lifecycle.LifecycleOwner;

import com.gyf.immersionbar.ImmersionBar;
import com.kkkkkn.readbooks.R;


public class BaseActivity extends AppCompatActivity  {
    private static final String TAG="BaseActivity";


    public void exitAll(){
        StackManager stackManager=StackManager.getInstance();
        stackManager.exitAllActivity();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        StackManager stackManager=StackManager.getInstance();
        stackManager.removeActivity(this);
    }

    @Override
    protected void onCreate(@Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

       /* Window window = getWindow();
        //添加Flag把状态栏设为可绘制模式
        window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
        window.getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN|View.SYSTEM_UI_FLAG_LAYOUT_STABLE);
        window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
        window.setStatusBarColor(Color.TRANSPARENT);
        //view不根据系统窗口来调整自己的布局
        ViewGroup mContentView = window.findViewById(Window.ID_ANDROID_CONTENT);
        View mChildView = mContentView.getChildAt(0);
        if (mChildView != null) {
            ViewCompat.setFitsSystemWindows(mChildView, false);
            ViewCompat.requestApplyInsets(mChildView);
        }

        //修改状态栏文字颜色
        View decor = window.getDecorView();
        decor.setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN | View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);
        // decor.setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN | View.SYSTEM_UI_FLAG_LAYOUT_STABLE);
*/
        //将当前activity加入栈
        StackManager stackManager=StackManager.getInstance();
        stackManager.addActivity(this);

        //沉浸式状态栏
        ImmersionBar.with(this)
                .statusBarDarkFont(true)
                .navigationBarDarkIcon(true)
                .barColor(R.color.toolbar)
                .init();
        Log.i(TAG, "onCreate:这是父类 "+this);

    }
}

File: app/src/main/java/com/kkkkkn/readbooks/view/activities/BaseActivity.kt
package com.kkkkkn.readbooks.view.activities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.viewbinding.ViewBinding
import com.gyf.immersionbar.ImmersionBar
import com.kkkkkn.readbooks.R

abstract class BaseActivity<T:ViewBinding>: AppCompatActivity() {
    lateinit var mViewBinding: T
    lateinit var tag:String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mViewBinding = getViewBinding()
        setContentView(mViewBinding.root)
        //设置tag名字
        tag = this.localClassName
        //将当前activity加入栈
        StackManager.instance?.addActivity(this)

        //沉浸式状态栏
        editStatusBar()
    }

    open fun editStatusBar(){
        ImmersionBar.with(this)
            .statusBarDarkFont(true)
            .navigationBarDarkIcon(true)
            .barColor(R.color.toolbar)
            .init()
    }

    abstract fun getViewBinding():T

    fun exitAll(){
        StackManager.instance?.exitAllActivity()
    }

    override fun onDestroy() {
        super.onDestroy()
        StackManager.instance?.removeActivity(this)
    }
}